lib/ui/playground/playground_view.dart

```diff
@@ -0,0 +1,247 @@
+import 'package:flutter/material.dart';
+
+class PlaygroundView extends StatefulWidget {
+  static const double _defaultFontSize = 40;
+  static const String _defaultTitle = 'Playground';
+  static const String _defaultAllWidgetsTitle = 'All widgets';
+  static const _defaultAppBarTitle = Text(
+    _defaultTitle,
+    style: TextStyle(
+      fontWeight: FontWeight.bold,
+      fontSize: _defaultFontSize,
+    ),
+  );
+
+  final List<Widget> widgetList;
+  final Color appBarColor;
+  final Widget appBarTitle;
+  final String allWidgetsText;
+
+  PlaygroundView(
+      {Key key,
+      @required this.widgetList,
+      this.appBarColor = const Color(0xFF00CD9F),
+      this.appBarTitle = _defaultAppBarTitle,
+      this.allWidgetsText = _defaultAllWidgetsTitle});
+
+  @override
+  _PlaygroundViewState createState() => _PlaygroundViewState();
+}
+
+class _PlaygroundViewState extends State<PlaygroundView> {
+  static const int _firstIndex = 0;
+  static const int _hardcodedItemSize = 1;
+  static const String _searchText = 'Search';
+
+  List<Widget> items = [];
+  TextEditingController editingController = TextEditingController();
+
+  @override
+  void initState() {
+    items.addAll(widget.widgetList);
+    super.initState();
+  }
+
+  @override
+  void dispose() {
+    items = null;
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      child: Column(
+        children: <Widget>[
+          Padding(
+            padding: const EdgeInsets.all(16.0),
+            child: _inputTextField(),
+          ),
+          Expanded(
+            child: _listView(),
+          ),
+        ],
+      ),
+    );
+  }
+
+  void _filterSearchResults(String query) {
+    if (query.isNotEmpty) {
+      var filteredItems = items
+          .where((widget) =>
+              widget.toString().toLowerCase().contains(query.toLowerCase()))
+          .toList();
+      setState(() {
+        items.clear();
+        items.addAll(filteredItems);
+      });
+    } else {
+      _resetList();
+    }
+  }
+
+  void _resetList() {
+    setState(() {
+      items.clear();
+      items.addAll(widget.widgetList);
+    });
+  }
+
+  ListView _listView() {
+    return ListView.builder(
+        shrinkWrap: true,
+        itemCount: items.length + _hardcodedItemSize,
```

Comment:
"size" makes me thing of a UI size, I wonder if that should be called _hardcodedItemCount